### ISSUE TITLE
Assigning security group by name works for EC2, not RDS.

### ISSUE THREAD

**Issue by davidjeddy** [terraform-providers/terraform-provider-aws#4316]
Hi there,

I was attempting to assign a security group to a RDS instance and kept receive an error about the SG not existing. Took ne about 4 hours to realize that for an RDS vpc_security_group_ids the value must be a list of SG IDs, whereas the EC2 vpc_security_group_ids will accept a list of names. This is inconsistent behavior and lead to my frustration.

### Terraform Version
Terraform v0.11.7

### Affected Resource(s)
provider.aws v1.15.0

### Terraform Configuration Files
RDS 
```hcl
resource "aws_db_instance" "wp_db" {
  allocated_storage    = 10
  storage_type         = "gp2"
  engine               = "mariadb"
  engine_version       = "10.2"
  instance_class       = "db.t2.micro"
  name                 = "wp_db"
  username             = "wp_db_user"
  password             = "wp_db_pass"
  parameter_group_name = "default.mariadb10.2"
  skip_final_snapshot  = true

  # list of security groups for the instance
  vpc_security_group_ids = [
    "${aws_security_group.wp_db_security_group.name}"
  ]

  tags {
    Name     = "wp"
    stage    = "demo"
    language = "sql"
    service  = "sql"
  }
}

```

RDS SG
```hcl
resource "aws_security_group" "wp_db_security_group" {
  name        = "wp_db_security_group"
  description = "Control access to the wp database server."
}

# TCP

# INGRESS

# allow ssh access from port 3306 (sql) from ${var.ec2_instance_wp_private_ips}/32
resource "aws_security_group_rule" "ingress_sql" {
  security_group_id = "${aws_security_group.wp_db_security_group.id}"
  type              = "ingress"
  cidr_blocks       = ["${var.ec2_instance_wp_private_ips[0]}/32"]
  protocol          = "tcp"
  from_port         = 3306
  to_port           = 3306
}
# EXGRESS

# allow reply traffic from the server to the internet on ephemeral ports
resource "aws_security_group_rule" "engress_sql" {
  security_group_id = "${aws_security_group.wp_db_security_group.id}"
  type              = "egress"
  cidr_blocks       = ["0.0.0.0/0"]
  protocol          = "all"
  from_port         = 1024
  to_port           = 65535
}
```

EC2
```hcl
# wp web server instance
resource "aws_instance" "wp" {
  ami = "${lookup(var.wp_images, var.region)}"

  # free tier instance
  instance_type = "t2.micro"

  # list of security groups for the instance
  vpc_security_group_ids = [
    "${aws_security_group.wp_security_group.name}"
  ]

  availability_zone = "${lookup(var.zones, var.region)}"

  key_name = "aws"

  # add a public IP address
  associate_public_ip_address = true

  root_block_device = {
    "delete_on_termination" = false
    "volume_type"           = "standard"
    "volume_size"           = 40
  }

  tags {
    Name     = "wp"
    stage    = "demo"
    language = "php"
    service  = "web"
  }
}
```

EC2 SG
```hcl
resource "aws_security_group" "wp_security_group" {
  name        = "wp_security_group"
  description = "Control the access to the wp server."
}

# TCP

# INGRESS

# allow ssh access from port 22 from variables.tfvars::var.local_ip
resource "aws_security_group_rule" "ingress_ssh" {
  security_group_id = "${aws_security_group.wp_security_group.id}"
  type              = "ingress"
  cidr_blocks       = ["${var.local_ip}/32"]
  protocol          = "tcp"
  from_port         = 22
  to_port           = 22
}

# allow http access on port 80 for all address/ports
resource "aws_security_group_rule" "ingress_http" {
  security_group_id = "${aws_security_group.wp_security_group.id}"
  type              = "ingress"
  cidr_blocks       = ["0.0.0.0/0"]
  protocol          = "tcp"
  from_port         = 80
  to_port           = 80
}

# allow https access on port 443 for all address/ports
resource "aws_security_group_rule" "ingress_https" {
  security_group_id = "${aws_security_group.wp_security_group.id}"
  type              = "ingress"
  cidr_blocks       = ["0.0.0.0/0"]
  protocol          = "tcp"
  from_port         = 443
  to_port           = 443
}

# EXGRESS

# allow reply traffic from the server to the internet on ephemeral ports
resource "aws_security_group_rule" "engress_reply" {
  security_group_id = "${aws_security_group.wp_security_group.id}"
  type              = "egress"
  cidr_blocks       = ["0.0.0.0/0"]
  protocol          = "all"
  from_port         = 1024
  to_port           = 65535
}

# ICMP

# INGRESS

# allow ICMP access from the internet
resource "aws_security_group_rule" "ingress_icmp" {
  security_group_id = "${aws_security_group.wp_security_group.id}"
  type              = "ingress"
  cidr_blocks       = ["0.0.0.0/0"]
  protocol          = "icmp"
  from_port         = 8
  to_port           = 0
}

# EGRESS

resource "aws_security_group_rule" "egress_icmp" {
  security_group_id = "${aws_security_group.wp_security_group.id}"
  type              = "egress"
  cidr_blocks       = ["0.0.0.0/0"]
  protocol          = "icmp"
  from_port         = 8
  to_port           = 0
}
```

### Debug Output
n/a

### Panic Output
n/a

### 'terraform apply' Output
https://gist.github.com/davidjeddy/10a304c9fb93dc1e9477aedeb99016df

### Expected Behavior
The security group should NOT be assigned tot he EC2 instance when using the NAME of the SG in the configuration of EC2.vpc_security_group_ids[].

### Actual Behavior
Able to use the SG name to assign to an EC2 instance; however, RDS does not behave this way.

### Steps to Reproduce
Create EC2 config.
Create SG config for EC2.
Create RDS (MariaDB) config
Create SG config for RDS.

Using the SG.name assign EC2 SG to EC2 instance config.
Using the SG.name assign RDS SG to RDS instance config.
Attempt to apply.

### Important Factoids
OSX 10.13.3 DEV env.

### References
n/a

**Comment by tomelliff** [comment]
@davidjeddy I can't reproduce your `aws_instance` using a security group name in the `vpc_security_group_ids` field. When I try it I get an error saying the security group can't be found. Do you have an EC2-Classic account?

Unfortunately AWS' API for EC2-Classic accounts seems to just handle names in a non breaking way while non EC2-Classic accounts created after the introduction of VPCs do require security group IDs to be passed.

I feel like the docs and the naming of the field are reasonably clear that you should be passing IDs rather than names here so not sure how we can best make it more obvious to users with an old style EC2-Classic account that this behaviour is slightly more accommodating here.

One option could be to add some validation on the `vpc_security_group_ids` field to make sure that they are all strings with a prefix of `sg-` which would technically be a breaking change for EC2-Classic users that are relying on being able to pass a security group name here.

**Comment by davidjeddy** [comment]
@tomelliff Unsure if it was a ec2 classic account as I no longer have access to the account. I do like the idea of adding the validation, as terraform is not yet a 1.x product release minor point releases are reasonable to have per [SemVer](https://semver.org/) expectations. I realize a lot of larger org's use TF so that may not be an option. How about half way? Throw a warning for the next 1/2 dozen minor point releases then sun-set the current behavior allowance for the more unified behavior of validating for `sg-*`?

**Comment by tomelliff** [comment]
Terraform is pre 1.x but the AWS provider is post 1.x so the change would have to be released as part of the 2.0.0 release if it was done.

Unfortunately I've not got access to an EC2 classic account to test the behaviour so I'd be uncomfortable raising the change without being able to fully test it.

**Comment by aeschright** [comment]
Closing per above request. Thanks @davidjeddy for reporting it!
